R/stats-factanal-tidiers.R

```diff
@@ -75,8 +75,8 @@ augment.factanal <- function(x, data, ...) {
   }
 
   # Place relevant values into a tidy data frame
-  tidy_df <- data.frame(.rownames = rownames(scores), data.frame(scores)) %>%
-    as_tibble()
+  tidy_df <- bind_cols(tibble(.rownames = rownames(scores)),
```

Comment:
You might be able to just do `tibble(.rownames = rownames(scores), scores)`?

Reply:
Your suggestion gives a different result (scores is all jammed into a single column).  How about this?

`tidy_df3 <- cbind(tibble(.rownames = rownames(scores)), scores)`

Here is a simple reprex that shows results of this line of code:

```
library(magrittr)
library(tibble)
x <- fit_reg
data = mtcars
scores <- x$scores
tidy_df <- data.frame(.rownames = rownames(scores), data.frame(scores)) %>% 
  as_tibble()
tidy_df2 <- tibble(.rownames = rownames(scores), scores)
tidy_df3 <- cbind(tibble(.rownames = rownames(scores)), scores)

str(tidy_df)
str(tidy_df2)
str(tidy_df3)
```
Here are the data.frames of:
1) what  I currently have in the code
2) your suggestion
3) my new suggestion
```
> str(tidy_df)
Classes ‘tbl_df’, ‘tbl’ and 'data.frame':	32 obs. of  4 variables:
 $ .rownames: Factor w/ 32 levels "AMC Javelin",..: 18 19 5 13 14 31 7 21 20 22 ...
 $ Factor1  : num  0.847 0.722 0.686 -0.866 -0.893 ...
 $ Factor2  : num  0.672 0.384 -0.592 -0.673 0.862 ...
 $ Factor3  : num  -0.2783 0.0246 -0.5644 -0.7666 -1.015 ...
> str(tidy_df2)
Classes ‘tbl_df’, ‘tbl’ and 'data.frame':	32 obs. of  2 variables:
 $ .rownames: chr  "Mazda RX4" "Mazda RX4 Wag" "Datsun 710" "Hornet 4 Drive" ...
 $ scores   : num [1:32, 1:3] 0.847 0.722 0.686 -0.866 -0.893 ...
  ..- attr(*, "dimnames")=List of 2
  .. ..$ : NULL
  .. ..$ : chr  "Factor1" "Factor2" "Factor3"
> str(tidy_df3)
'data.frame':	32 obs. of  4 variables:
 $ .rownames: chr  "Mazda RX4" "Mazda RX4 Wag" "Datsun 710" "Hornet 4 Drive" ...
 $ Factor1  : num  0.847 0.722 0.686 -0.866 -0.893 ...
 $ Factor2  : num  0.672 0.384 -0.592 -0.673 0.862 ...
 $ Factor3  : num  -0.2783 0.0246 -0.5644 -0.7666 -1.015 ...
```

Reply:
Okay let's keep what's there then.